static/app/views/insights/sessions/components/insightLayoutContext.tsx

```diff
@@ -0,0 +1,68 @@
+import type {ReactNode} from 'react';
+import {createContext, useCallback, useContext, useEffect, useState} from 'react';
+
+import localStorage from 'sentry/utils/localStorage';
+import type {DomainView} from 'sentry/views/insights/pages/useFilters';
+import {
+  type CHART_MAP,
+  DEFAULT_LAYOUTS,
+  PAGE_CHART_OPTIONS,
+} from 'sentry/views/insights/sessions/components/settings';
+
+type TChart = keyof typeof CHART_MAP;
+type TCharts = readonly TChart[];
+interface TContext {
+  chartOptions: TCharts;
+  chartsByIndex: TCharts;
+  onChange: (index: number, chart: TChart) => void;
+  view: DomainView;
+}
+
+const defaultContext: TContext = {
+  chartOptions: [],
+  chartsByIndex: [],
+  onChange: () => {},
+  view: 'frontend',
```

Comment:
Choosing `frontend` here is a little spooky, if we end up changing what the default view is, will we need to update this? Since that value comes from a prop, does it need a default?

Reply:
ah, view isn't needed!